scripts/zones/Grand_Palace_of_HuXzoi/Zone.lua

```diff
@@ -105,4 +110,20 @@ function onEventFinish(player,csid,option)
     if (csid >0x0095 and csid < 0x00A0) then
         player:setVar("Hu-Xzoi-TP",0);
     end
+end;
+
+-----------------------------------
+-- onGameHour
+-----------------------------------
+
+function onGameHour(npc, mob, player)
+	local VanadielHour = VanadielHour();

```

Comment:
onGameHour only called for Zones (zone.lua), not for NPCs or Mobs.

See luautils.cpp line 2778 for declaration of onGameHour